Automattic/newspack-migration-tools
# Add fancy getter to migrationobjectclass and some tests

This is a less destructive version of https://github.com/Automattic/newspack-migration-tools/pull/18 

It still adds in the Symfony PropertyAccess and uses that in the getter (and the setter for validation). Look at the tests for how to use. If we prefer to implement the [ArrayAccess](https://www.php.net/manual/en/class.arrayaccess.php) interface instead we can do that too – it will just require some more work if we need to support both `object|array`. If we only support array then that might be the better option.

* I made the `$pointer_to_id` not optional. 
* I ran `phpcbf` to stop `phpcs` from freaking out – so maybe review this with whitespace changes off :)
* There are still `phpcs` errors in other code I didn't touch – we should fix them so we can go green.

Note that that library requires PHP 8.3, so this also adds a requirement for that. 

The tests only really test what Symfony probably already does, but it is for us to have a starting point for tests.

## How to test
* Install the tests. [See README](https://github.com/Automattic/newspack-migration-tools/blob/trunk/README.md#tests)
* Run `composer run-script phpunit -- --filter TestMigrationObject`
* Play around with it an look at the code
